### PR TITLE
[improvement](cloud) reduce call for partition get visible version

### PR DESCRIPTION
**Pull request by yujun777**
For cloud,  partition's get visible version need a rpc.  So need to reduce call partition visible version.

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Review by deardeng — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java [hunk 1]
```diff
@@ -1781,13 +1781,14 @@ private Status allTabletCommitted(boolean isReplay) {
 
                     // update partition visible version
                     part.updateVersionForRestore(entry.getValue());
+                    long visibleVersion = part.getVisibleVersion();
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:1784]: We need to use `org.apache.doris.cloud.catalog.CloudPartition#getSnapshotVisibleVersion(java.util.List<org.apache.doris.cloud.catalog.CloudPartition>)`
to get versions in batch for efficiency

Make an in-memory temporary version-cache will do

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapTableSink.java [hunk 2]
```diff
@@ -625,6 +625,7 @@ public List<TOlapTableLocationParam> createLocation(OlapTable table) throws User
         Multimap<Long, Long> allBePathsMap = HashMultimap.create();
         for (Long partitionId : partitionIds) {
             Partition partition = table.getPartition(partitionId);
+            long visibleVersion = partition.getVisibleVersion();
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OlapTableSink.java:628]: ditto, make it in batch

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java (hunk 1 above)
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:1784]: restore 是恢复备份， cloud 甚至不支持这个功能

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**